src/parser/index.js

```diff
@@ -1,11 +1,41 @@
+import type { Options } from "../options";
+import * as N from "../types";
 import { reservedWords } from "../util/identifier";
 import { getOptions } from "../options";
 import Tokenizer from "../tokenizer";
 
-export const plugins = {};
+export const plugins: { [key: string]: (superClass: Class<Parser>) => Class<Parser> } = {};
 
 export default class Parser extends Tokenizer {
-  constructor(options: Object, input: string) {
+  options: Options;
+  inModule: boolean;
+  plugins: { [key: string]: boolean };
+  filename: ?string;
+
+  /**
+   * Add methods to the parser prototype.
+   * This *must* be used like so:
+   *
+   * Parser.mixin(class extends Parser {
+   *   myMethod() { ... }
+   *   ...other methods...
+   * });
+   *
+   * Mixin methods you want visible from the outside should be declared in `parserTypes.js`.
+   */
+  static mixin(mixedIn: Class<Parser>): void {
+    const proto: any = Parser.prototype;
+    const mixedProto: any = mixedIn.prototype;
+    for (const key of Object.getOwnPropertyNames(mixedProto)) {
+      if (key === "constructor")
+        continue;
+      if (proto[key])
+        throw new Error(`Can't mixin '${key}', already defined`);
+      proto[key] = mixedProto[key];
```

Comment:
The copied keys will be enumerable, but they are not when defined in the `class` expression.

On the other hand, before the change to `class` expression, they were enumerable anyway, so it might not matter.